File: src/main/kotlin/com/marcel/baumgardt/service/ShowingService.java
package com.marcel.baumgardt.service;

import com.marcel.baumgardt.model.dto.ShowingDatesResponse;
import com.marcel.baumgardt.model.dto.UpdateShowingDatesRequest;
import com.marcel.baumgardt.model.dto.UpdateShowingPriceRequest;
import com.marcel.baumgardt.model.dto.UpdateShowingResponse;

public interface ShowingService {

    UpdateShowingResponse updatePrice(UpdateShowingPriceRequest request);

    UpdateShowingResponse updateDates(UpdateShowingDatesRequest request);

    ShowingDatesResponse getDates(Long cinemaId, Long movieId);
}

File: src/main/kotlin/com/marcel/baumgardt/service/interfaces/ShowingService.kt
package com.marcel.baumgardt.service.interfaces

import com.marcel.baumgardt.model.dto.request.UpdateShowingDatesRequest
import com.marcel.baumgardt.model.dto.request.UpdateShowingPriceRequest
import com.marcel.baumgardt.model.dto.response.ShowingDatesResponse
import com.marcel.baumgardt.model.dto.response.UpdateShowingResponse

interface ShowingService {
    fun updatePrice(request: UpdateShowingPriceRequest): UpdateShowingResponse
    fun updateDates(request: UpdateShowingDatesRequest): UpdateShowingResponse
    fun getDates(cinemaId: Long, movieId: Long): ShowingDatesResponse
}